### ISSUE TITLE
null user in validateCredentials

### ISSUE THREAD

**Issue by arose-hat** [Adldap2/Adldap2-Laravel#178]
I've got Laravel working with Ldap, I can authenticate manually and pull back AD info.

When I try to login normally it authenticates me and everything works as far as validateCredentials in AdldapAuthUserProvider.php.

```
public function validateCredentials(Authenticatable $user, array $credentials)
    {
        // Check if we already have an authenticated AD user.
        if ($this->user instanceof User) {
```

$this->user is null

$user is of class App\User:

``` php
User {#286 ▼
  #fillable: array:4 [▼
    0 => "name"
    1 => "email"
    2 => "password"
    3 => "username"
  ]
  #hidden: array:2 [▶]
  #connection: null
  #table: null
  #primaryKey: "id"
  #keyType: "int"
  #perPage: 15
  +incrementing: true
  +timestamps: true
  #attributes: array:5 [▼
    "email" => "**************"
    "password" => "************************"
    "name" => "*******************"
    "username" => "*********************"
    "adldapUser" => User {#278 ▶}
  ]
  #original: []
  #relations: []
  #visible: []
  #appends: []
  #guarded: array:1 [▶]
  #dates: []
  #dateFormat: null
  #casts: []
  #touches: []
  #observables: []
  #with: []
  #morphClass: null
  +exists: false
  +wasRecentlyCreated: false
}
```

Any ideas where I might be going wrong?

**Comment by warlord0** [comment]
Laravel 5.3?
if so replace the
```php
protected $username = 'username';
```
with
```php
public function username()
{
    return 'username';
}
```
in your LoginController.php

**Comment by stevebauman** [comment]
Hi @arose-hat,

This is a strange issue. Do you receive any exceptions at all? And do you perhaps have fallback enabled?

**Comment by arose-hat** [comment]
Laravel 5.2.43

Fallback is enabled. I've disabled it and I just get thrown back to the login page with "These credentials do not match our records". No exceptions.

**Comment by arose-hat** [comment]
As users were authenticating, but nothing was showing up in the db I forced a model save in a part of the code that does get run. This gave me an error 'Object of class Adldap\Models\User could not be converted to string'.

I've updated to the latest version and set bind_user_to_model to false and now logins are working. With this set to true I get the above error. Is this the expected behaviour? I though that the sync attributes governed what ad info got stored, but it's also trying to dump the whole ad user to the db.
